keras/backend/tensorflow_backend.py

```diff
@@ -1486,6 +1486,15 @@ def l2_normalize(x, axis):
     return tf.nn.l2_normalize(x, dim=axis)
 
 
+def in_top_k(predictions, targets, k):
+    '''Given tensors representing predictions and targets, evaluates whether the
+    targets are within the top k predictions
+    '''
+    y_pred = reshape(predictions, [1, shape(predictions)[0]])
+    y_target = reshape(argmax(targets), [1])
+    return cast(any(tf.nn.in_top_k(y_pred, y_target, k)), 'float32')
```

Comment:
`any` seems it doesn't belong here. You'll want this to return an array. Additionally, I would prefer not casting to `float32` and use the native TF `bool` type instead (i.e. this function should be a thin wrapper over `tf.nn.in_top_k`). The fact that Theano does not have a bool type has to be dealt with in the Theano backend and should not affect the TensorFlow backend.